src/frontend/src/binder/bind_context.rs

```diff
@@ -109,6 +110,36 @@ impl BindContext {
             .into()),
         }
     }
+
+    /// Merges two `BindContext`s which are adjacent. For instance, the `BindContext` of two
+    /// adjacent cross-joined tables.
+    pub fn merge_context(&mut self, other: Self) -> Result<()> {
```

Comment:
add unit tests for this function?

Reply:
well yes, but I have to say that the entire binder module has basically 0 unit test coverage.